How-to-find-the-commit-which-introduced-a-bug.md

```diff
@@ -0,0 +1,186 @@
+2. **Depending on how far back you want to go, checkout one of the following:**
+    * **This commit from Dec 2022 &lt;[9a334e9bde1d3d10e3b69dcd461d3e649733b0c0](https://github.com/oppia/oppia/commit/9a334e9bde1d3d10e3b69dcd461d3e649733b0c0)>, which is when the Python version was last updated – though this might not work if you are using Docker.**
+    * **This commit from Aug 2023 &lt;<code>[dc333e4](https://github.com/oppia/oppia/commit/dc333e4e25dc72e22910cb6f8ef32ae652a29dad)</code>>, which is when Docker was first usable.</strong>
+    * <strong>The latest release commit (you can find the hash and release date [here](https://github.com/oppia/oppia/releases)), if you want to see whether the error happened since the last release;</strong>
+3. <strong>If the issue is <span style="text-decoration:underline;">not</span> reproducible on that commit then go ahead with the approach provided below.</strong>
+4. <strong>If the issue is reproducible on that commit, report that info on the issue thread or group chat as appropriate. Include a link to the commit and mention the month/year it was made.</strong>
+
+<!-- <span style="text-decoration:underline; font-size: 20px;">Git Blame</span> -->
+## Git Blame
+
+**`git blame` is a powerful Git command that helps you identify who last modified each line of a file. It's a useful tool for understanding the history of changes in a file and finding out who made specific modifications.**
+
+How to use `git blame` to find bad commits:
+
+
+1. Based on console errors or analysis of the code, find out which line(s) of code are causing problems.
+2. Run `git blame` for that specific file and see if the line was recently introduced to the file.
+3. Once the commit is obtained from the “blame” view, checkout that commit
+    ```bash
+    git checkout <commit-id>
+    ```
+    and see if the error is present. If yes, then checkout its parent commit (one commit earlier – you can find out what this is with `git log`).
+
+4. If the parent commit doesn’t have errors, then you’ve confirmed that the error is due to that commit through which the lines are added to the codebase. Congratulations! Mention this info in the issue thread and include a link to the problematic commit.
+
+## Using GitHub Website
+    
+
+Using the GitHub website, you can view the blame information for a file directly in the GitHub interface. Let’s try and use this feature on a file (contact-page.component.html) in the Oppia Repository.
+
+If you want a **guided Practical** go [here](https://app.tango.us/app/workflow/Using-Blame-with-github-Website--f48fa87dc2e749e2af2dbd10281f4ec2), and then click on "**Guide me**".
+
+
+Let’s start: 
+
+
+1. **Open the GitHub repository in your web browser.**
+
+![Screenshot of the home page](images/findCommitWhichIntroducedBug/image2.png)
+
+
+2. **Navigate to the file(In this example we need to navigate to the core/templates/pages/contact-page/contact-page.component.html)**
```

Comment:
Navigate to the file. (In this example, we need to ...)


Btw, do you actually need to bold every single line? It'll make it harder to read -- let's instead reserve bolding for specific parts of the text we want to emphasize.

Reply:
Done